src/jit/codegencommon.cpp

```diff
@@ -3766,31 +3766,52 @@ void            CodeGen::genFnPrologCalleeRegArgs(regNumber xtraReg,
         return;
     }
 #endif
+    unsigned    argMax;            // maximum argNum value plus 1, (including the RetBuffArg)
```

Comment:
If a method has 3 args (and no return buffer) then argMax is 3 and valid indexes are 0,1,2
If a method has a return buffer on Arm64 then argMax gets set to 8 and valid index are 0-7
The table can always have unused entries, for example if an architecture always increments the arg register number then with a mix of float and integer args you could have:
r0, r2 and r4 as valid integer arguments with argMax as 4 
and f1, f3 and f5 and valid floating point arguments with argMax as 5
If a ARM64 fixed  return buffer was also present then the first one would become:
r0, r2, r4 and r8 as valid integer arguments with argMax as 8